src/compiler/moduleNameResolver.ts

```diff
@@ -665,16 +665,23 @@ namespace ts {
                 }
             }
 
-            switch (moduleResolution) {
-                case ModuleResolutionKind.NodeJs:
-                    result = nodeModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
-                    break;
-                case ModuleResolutionKind.Classic:
-                    result = classicNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
-                    break;
-                default:
-                    return Debug.fail(`Unexpected moduleResolution: ${moduleResolution}`);
+            try {
+                if (etwLogger) etwLogger.logStartResolveModule(moduleName /* , containingFile, ModuleResolutionKind[moduleResolution]*/);

```

Comment:
Rather than all of these conditional `if (etwLogger)` statements, can we make a `Logger` which _always_ exists, but is set to a null logger (every function is `noop`) unless `etwLogger` is present?

Reply:
That seems much nicer than what we have now, but would it still be worthwhile if we had Elvis?

Reply:
Hmmmm, I think so.
```
logger.logStartWhatever();
```
and
```
logger?.logStartWhatever();
```
aren't _that_ different, so were it available right now I probably wouldn't complain, but I still prefer the first, personally (specially since the logging is essentially unconditional, logically, it just the _logger_ might not be there).

Reply:
Also I think unless absolutely necessary we should avoid adding try catch blocks.

Reply:
I believe (but don't know) that Bill added these because unbalanced ETW events are much more troublesome than unbalanced log entries.

Reply:
What's the rationale for avoiding try-catch blocks? Does it make a difference that we are only introducing try-finally blocks without any catching?